x-pack/plugin/sql/qa/jdbc/src/main/java/org/elasticsearch/xpack/sql/qa/jdbc/PreparedStatementTestCase.java

```diff
@@ -65,25 +79,26 @@ public void testSupportedTypes() throws SQLException {
                 sql.add("?");
             }
             try (PreparedStatement statement = connection.prepareStatement(sql.toString())) {
-                statement.setString(1, stringVal);
-                statement.setInt(2, intVal);
-                statement.setLong(3, longVal);
-                statement.setFloat(4, floatVal);
-                statement.setDouble(5, doubleVal);
-                statement.setNull(6, JDBCType.DOUBLE.getVendorTypeNumber());
-                statement.setBoolean(7, booleanVal);
-                statement.setByte(8, byteVal);
-                statement.setShort(9, shortVal);
-                statement.setBigDecimal(10, bigDecimalVal);
-                statement.setTimestamp(11, timestampVal);
-                statement.setTimestamp(12, timestampVal, calendarVal);
-                statement.setDate(13, dateVal);
-                statement.setDate(14, dateVal, calendarVal);
-                statement.setTime(15, timeVal);
-                statement.setTime(16, timeVal, calendarVal);
-                statement.setObject(17, calendarVal);
-                statement.setObject(18, utilDateVal);
-                statement.setObject(19, localDateTimeVal);
+                int index = 1;
+                statement.setString(index++, stringVal);
+                statement.setInt(index++, intVal);
+                statement.setLong(index++, longVal);
+                statement.setFloat(index++, floatVal);
+                statement.setDouble(index++, doubleVal);
+                statement.setNull(index++, JDBCType.DOUBLE.getVendorTypeNumber());
+                statement.setBoolean(index++, booleanVal);
+                statement.setByte(index++, byteVal);
+                statement.setShort(index++, shortVal);
+                statement.setBigDecimal(index++, bigDecimalVal);
+                statement.setTimestamp(index++, timestampVal);
+                statement.setTimestamp(index++, timestampVal, calendarVal);
+                statement.setDate(index++, dateVal);
+                statement.setDate(index++, dateVal, calendarVal);
+                statement.setTime(index++, timeVal);
+                statement.setTime(index++, timeVal, calendarVal);
+                statement.setObject(index++, calendarVal);
+                statement.setObject(index++, utilDateVal);
+                statement.setObject(index++, localDateTimeVal);
 
```

Comment:
> I'm not a fan of this change.

FTR, this (and similar) was part of the UL PR. (just to give the credit to the wrong-doer :-) )

Asking just for future, this PR should take its own path, of course.

> The index++ doesn't buy us anything - the method calls are still there but the actual value is not hidden behind index++.

The change was prompted by the small annoyance caused every time a new insertion is made (not often, admittedly): the PR has to change all the lines above the insertion. Which generates this type of diff every time, instead of a one liner.

Reply:
Fair enough however how many times are we going to introduce a new type?
I'm fine leaving it as it is.